### PR TITLE
do not throttle admin apis

### PR DESCRIPTION
**Pull request by calellowitz**
## Technical Summary

https://dimagi-dev.atlassian.net/browse/SAAS-14178

This should fix a problem where our internal use of APIs for salesforce was being rate limited by removing throttling for admin apis. They do not fit naturally into our existing throtlling since they do not map to any domain and they are only accessible to superusers so there is not a real risk of abuse.

I considered making a separate `NoThrottleMixin` that these `Meta` classes could inherit from, to make the change a bit more explicit (and easier to replicate in the future), but it did not seem worth it. Happy to take that approach if it makes sense to others, I don't love the need to manually set the throttle on each class, although each had custom `Meta` already so much of this was already manual, and those with shared `Meta` like the accounting ones were able to be fixed in one place.

### Safety story

Limited to superusers, so low risk of unexpected use, and easy to resolve if it happens.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by mjriley — CHANGES_REQUESTED** [review]
I liked your thoughts about wishing you could implicitly specify the throttle behavior by setting admin authentication. Is there a reason we can't just do this by creating an `AdminMeta` base class that provides reasonable defaults for any admin APIs? Looking through our existing classes, they mostly inherit from `CustomResourceMeta` only for the `authorization` and `default_format` fields (and sometimes `serializer`). Another option might be to use tastypie's `ResourceOptions` class as the base class, as its comment suggests it provides 'sane defaults'. I think leaving the fixes as is, will, as you mentioned, likely lead to someone forgetting to set the correct Throttle for a new Admin API.

Did we overlook `MaltResource`'s Meta class? That one also seems to be listed in the `ADMIN_API_LIST`, but it appears like it will still maintain its current throttling behavior.

Finally, are sure this unintended throttling is restricted to admin functionality? If we were to implement API functions to allow users to check their login history, for example, that would likely be scoped to an empty domain and run into the same issues. Any safeguards for that?

**Comment by calellowitz** [comment]
> I liked your thoughts about wishing you could implicitly specify the throttle behavior by setting admin authentication. Is there a reason we can't just do this by creating an `AdminMeta` base class that provides reasonable defaults for any admin APIs? Looking through our existing classes, they mostly inherit from `CustomResourceMeta` only for the `authorization` and `default_format` fields (and sometimes `serializer`). Another option might be to use tastypie's `ResourceOptions` class as the base class, as its comment suggests it provides 'sane defaults'. I think leaving the fixes as is, will, as you mentioned, likely lead to someone forgetting to set the correct Throttle for a new Admin API.

Yeah, somehow I saw all the customization in each one and forgot that changing the base class would not prevent that. I added a new class for all of them except the web user one, since that one did not inherit from `CustomResourceMeta` and functions differently enough from the other APIs, that having it be a little more explicit seemed reasonable.

> Did we overlook `MaltResource`'s Meta class? That one also seems to be listed in the `ADMIN_API_LIST`, but it appears like it will still maintain its current throttling behavior.

Yep, not sure how I missed that one, but fixed.

> Finally, are sure this unintended throttling is restricted to admin functionality? If we were to implement API functions to allow users to check their login history, for example, that would likely be scoped to an empty domain and run into the same issues. Any safeguards for that?

I think it certainly is for now. It is plausible that we will implement non domain based APIs in the future, although I don't think we have any plans to, but I think it is fine to punt on how we handle throttling for those until then.

File: corehq/apps/api/resources/v0_5.py [hunk 1]
```diff
@@ -302,6 +303,7 @@ class Meta(WebUserResource.Meta):
         authentication = AdminAuthentication()
         detail_allowed_methods = ['get']
         list_allowed_methods = ['get']
+        throttle = BaseThrottle()
```
Comment (mjriley) [inline corehq/apps/api/resources/v0_5.py:306]: If the intention of `AdminResourceMeta` is that any admin Meta inherits from it, this Meta is now pretty confusing. It's a class that begins with 'Admin', is listed in `ADMIN_API_LIST`, but doesn't inherit from `AdminResourceMeta`?  I'd imagine this _could_ inherit from `AdminResourceMeta` with no issues, but for the sake of getting this out, I can understand not experimenting with it -- I'd just want to add some kind of comment explaining that why it isn't inheriting from `AdminResourceMeta`.

**Review by mjriley — APPROVED** [review]
Changes look good. I'm still a bit concerned about `AdminWebUserResource`, but not in a way that would prevent this from going out right now.

File: corehq/apps/api/resources/v0_5.py (hunk 1 above)
Comment (calellowitz) [reply corehq/apps/api/resources/v0_5.py:306]: Do you mean to inherit from both `AdminResourceMeta` and `WebUserResource.Meta`?

Comment (mjriley) [reply corehq/apps/api/resources/v0_5.py:306]: I see your point. I don't want to encourage multiple inheritance, and I see what feels like an unnecessary inheritance hierarchy as is (WebUserResource relies on v0_1.WebUserResource, which relies on UserResource, which relies on CustomResourceMeta). The only thing we get out of this inheritance hierarchy that `AdminMeta` doesn't provide is `object_class` and `resource_name`, which don't seem like the kind of thing we need inheritance for.

**Review by mjriley — APPROVED** [review]
Appreciate the changes. Looks ready to go

**Pull request merged**